src/Core/Batch/BatchRunner.php

```diff
@@ -0,0 +1,183 @@
+     * @param SubmitItemInterface $submitter [optional] The submitter
+     *        object to use. **Defaults to** null.
+     */
+    public function __construct(
+        ConfigStorageInterface $configStorage = null,
+        SubmitItemInterface $submitter = null
+    ) {
+        if ($configStorage === null || $submitter === null) {
+            if ($this->isSysvIPCLoaded() && $this->isDaemonRunning()) {
+                $configStorage = new SysvConfigStorage();
+                $submitter = new SysvSubmitter();
+            } else {
+                $configStorage = InMemoryConfigStorage::getInstance();
+                $submitter = $configStorage;
+            }
+        }
+        $this->configStorage = $configStorage;
+        $this->submitter = $submitter;
+        $this->loadConfig();
+    }
+
+    /**
+     * Register a job for batch execution.
+     *
+     * @param string $identifier Unique identifier of the job.
+     * @param callable $func Any Callable except for Closure. The callable
+     *        should accept an array of items as the first argument.
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type int $batchSize The size of the batch.
+     *     @type float $callPeriod The period in seconds from the last execution
+     *                 to force executing the job.
+     *     @type int $workerNum The number of child processes. It only takes
+     *               effect with the {@see \Google\Cloud\Core\BatchDaemon}.
+     *     @type string $bootstrapFile A file to load before executing the
+     *                  job. It's needed for registering global functions.
+     * }
+     * @return boolean true on success, false on failure
+     * @throws \InvalidArgumentException When receiving a Closure.
+     */
+    public function registerJob($identifier, $func, $options = [])
+    {
+        if ($func instanceof \Closure) {
+            throw new \InvalidArgumentException('Closure is not allowed');
+        }
+        // Always work on the latest data
+        $result = $this->configStorage->lock();
+        if ($result === false) {
+            return false;
+        }
+        $this->config = $this->configStorage->load();
+        $this->config->registerJob($identifier, $func, $options);
+
+        $result = $this->configStorage->save($this->config);
+        if ($result === false) {
+            return false;
+        }
+        $this->configStorage->unlock();
+        return true;
+    }
+
+    /**
+     * Submit an item.
+     *
+     * @param string $identifier Unique identifier of the job.
+     * @param mixed $item It needs to be serializable.
+     *
+     * @return boolean true on success, false on failure
+     */
+    public function submitItem($identifier, $item)
+    {
+        $job = $this->getJobFromId($identifier);
+        if ($job === null) {
+            throw new \RuntimeException(
+                'The identifier does not exist: ' . $identifier
```

Comment:
In other cases you used `"The identifier does not exist: $identifier"`, I like that style if possible here as well